endtoend/evaluators/slashing.go

```diff
@@ -169,14 +170,105 @@ func insertDoubleAttestationIntoPool(conns ...*grpc.ClientConn) error {
 			Data:            attData,
 			Signature:       privKeys[committee[i]].Sign(signingRoot[:]).Marshal(),
 		}
-		for _, conn := range conns {
-			client := eth.NewBeaconNodeValidatorClient(conn)
-			_, err = client.ProposeAttestation(ctx, att)
-			if err != nil {
-				return err
-			}
+		client := eth.NewBeaconNodeValidatorClient(conns[0])
+		_, err = client.ProposeAttestation(ctx, att)
+		if err != nil {
+			return err
 		}
 		slashedIndices = append(slashedIndices, committee[i])
 	}
 	return nil
 }
+
+func proposeDoubleBlock(conns ...*grpc.ClientConn) error {
+	conn := conns[0]
```

Comment:
what happen if `len(conns) != 1` is empty?  Probably best to check the length and return error

Reply:
It will always be something here, the test would fail much earlier elsewhere if it wasn't.

Reply:
Agree.Bbut better to be defensive. Someone brand new can be writing an e2e test and use `proposeDoubleBlock` but 100% unaware `conns[0]` vs `[1]`, `[2]` will be used

Reply:
Also, why pass in `conns`, why do we need a list? why not just use `conn`?

Reply:
Putting the check here would be very odd, these style of tests follow an interface for all of E2E. Only `conns[0[` is used here but in the dozen or so other E2E tests, the other conns are used. Using multiple conns here just adds little benefit. E2E is designed with the assumption that there will always be at least 1 beacon node, because if there isn't, the test wouldn't progress.

Reply:
Tests are at least run with 1 node, so pinging for any chain data that shouldn't be node dependent is done with that node.

Reply:
Also, only using one beacon node here helps test `--broadcast-slashing`, its why I removed the usage of other nodes from `injectDoubleVote`.

Reply:
I see. I got confused. Perhaps we should add a comment to mention only con0 is explicitly used but input takes in a list of conns.

Reply:
Sure thing, that sounds good :+1: